### PR TITLE
"Add to Calendar" creates event with description instead of empty description

### PR DESCRIPTION
**Pull request by lad6558**
Previously, when click on "add to calendar", it creates an event with no descriptions. This feature uses the email html to populate the event description. 

Implementation-wise, it copies the "removeArtifact" util function from the daemon repo in an attempt to remove extra htmls. It doesn't completely remove all the extra line breaks as great as the daemon does (the events on the shared google calendar have great descriptions), but because frontend only has access to `SerializableEvent` (i.e. without text) I feel like this is the easiest way we can do it without shifting the architecture too much.

### CODE REVIEW

**Comment by lad6558** [comment]
This does not look great, but it is better than no description

image

File: app/util.ts [hunk 1]
```diff
@@ -1,11 +1,43 @@
 import { type } from "os";
 
 export const getLocationLink = (content: string) => {
-    const isMITLocationTest = new RegExp(".*-.*\\d$");
-    const isLocation = isMITLocationTest.test(content)
-    return isLocation ? `https://whereis.mit.edu/?go=${content}` : ""
-}
+  const isMITLocationTest = new RegExp(".*-.*\\d$");
+  const isLocation = isMITLocationTest.test(content);
```
Comment (gabrc52) [inline app/util.ts:5]: It would be best to run the linter in a separate commit.

File: app/util.ts [hunk 2]
```diff
@@ -1,11 +1,43 @@
...
 export const compareIgnoreCase = (str1: string, str2: string) => {
-    return str1.toUpperCase() == str2.toUpperCase();
-}
\ No newline at end of file
+  return str1.toUpperCase() == str2.toUpperCase();
+};
+
+function removeBase64(input: string) {
+  const startKeyword = ";base64,";
+  const start = input.indexOf(";base64,");
```
Comment (gabrc52) [inline app/util.ts]: nit: use startKeyword here

File: app/util.ts [hunk 3]
```diff
@@ -1,11 +1,43 @@
...
+};
 
 export const compareIgnoreCase = (str1: string, str2: string) => {
-    return str1.toUpperCase() == str2.toUpperCase();
-}
\ No newline at end of file
+  return str1.toUpperCase() == str2.toUpperCase();
+};
+
+function removeBase64(input: string) {
```
Comment (gabrc52) [inline app/util.ts]: Is this actually needed? Presumably the email has already been parsed and the email only contains the HTML.

**Comment by gabrc52** [comment]
From what I can tell, the functions you copied are designed for cleaning the *plaintext* version of an email, but you are sometimes calling that on the *HTML* version of an email.

Many Outlook emails seem to have something like this `<style type="text/css" style="display:none;"> P {margin-top:0;margin-bottom:0;} </style>`

and it seems like this style tag is getting added into the text

image

and what you intended to do from the function names (such as removing redundant line breaks) isn't actually being done

Could you please test locally with a few example emails/events?

**Review by gabrc52 — CHANGES_REQUESTED** [review]
From what I can tell, the functions you copied are designed for cleaning the *plaintext* version of an email, but you are sometimes calling that on the *HTML* version of an email.

It seems like in practice, a lot of the emails here are actually HTML emails, not plaintext.

image

Many Outlook emails seem to have something like this `<style type="text/css" style="display:none;"> P {margin-top:0;margin-bottom:0;} </style>`

and it seems like this style tag is getting added into the text in the calendar

image

and what you intended to do from the function names (such as removing redundant line breaks) isn't actually being done

Could you please test locally with a few example emails/events and make sure it works as intended?

**Comment by lad6558** [comment]
Hey Gabe, thanks for the constructive review and taking the effort to do local testing. I've used a better library to do html-to-text and tested on a few events with no problem. Would you like to re-review?

(the linter thing was applied automatically on save, so couldn't help with that)

**Review by gabrc52 — APPROVED** [review]

**Pull request merged**